examples/images/cifar10/train_cifar10_ddp.py

```diff
@@ -156,46 +154,48 @@ def train(rank, total_num_gpus, argv):
-                    loss = torch.mean((vt - ut) ** 2)
-                    loss.backward()
-                    torch.nn.utils.clip_grad_norm_(net_model.parameters(), FLAGS.grad_clip)  # new
-                    optim.step()
-                    sched.step()
-                    ema(net_model, ema_model, FLAGS.ema_decay)  # new
-
-                    # sample and Saving the weights
-                    if FLAGS.save_step > 0 and global_step % FLAGS.save_step == 0:
-                        generate_samples(
-                            net_model, FLAGS.parallel, savedir, global_step, net_="normal"
-                        )
-                        generate_samples(
-                            ema_model, FLAGS.parallel, savedir, global_step, net_="ema"
-                        )
-                        torch.save(
-                            {
-                                "net_model": net_model.state_dict(),
-                                "ema_model": ema_model.state_dict(),
-                                "sched": sched.state_dict(),
-                                "optim": optim.state_dict(),
-                                "step": global_step,
-                            },
-                            savedir + f"{FLAGS.model}_cifar10_weights_step_{global_step}.pt",
-                        )
+            for x1, _ in tqdm.tqdm(dataloader, total=len(dataloader)):
+                global_step += 1
+
+                optim.zero_grad()
+                x1 = x1.to(rank)
+                x0 = torch.randn_like(x1)
+                t, xt, ut = FM.sample_location_and_conditional_flow(x0, x1)
+                vt = net_model(t, xt)
+                loss = torch.mean((vt - ut) ** 2)
+                loss.backward()
+                torch.nn.utils.clip_grad_norm_(
+                    net_model.parameters(), FLAGS.grad_clip
+                )  # new
+                optim.step()
+                sched.step()
+                ema(net_model, ema_model, FLAGS.ema_decay)  # new
+
+                # sample and Saving the weights
+                if FLAGS.save_step > 0 and global_step % FLAGS.save_step == 0:
+                    generate_samples(
+                        net_model, FLAGS.parallel, savedir, global_step, net_="normal"
+                    )
+                    generate_samples(
+                        ema_model, FLAGS.parallel, savedir, global_step, net_="ema"
+                    )
+                    torch.save(
+                        {
+                            "net_model": net_model.state_dict(),
+                            "ema_model": ema_model.state_dict(),
+                            "sched": sched.state_dict(),
+                            "optim": optim.state_dict(),
+                            "step": global_step,
+                        },
+                        savedir
+                        + f"{FLAGS.model}_cifar10_weights_step_{global_step}.pt",
+                    )
```

Comment:
This chunk seems to change a lot but it actually only modifies where `x1` is read. The chunk comes from the indentation change as we do not need `step_pbar` now.